src/components/micro-frame-sse/parser.ts

```diff
@@ -0,0 +1,86 @@
+interface SSE {
+  lastEventId: string;
+  data: string;
+  type?: string;
+}
+export default (read?: (param: SSE) => [string, string, boolean?]) =>
+  function parser(
+    iterator: AsyncIterator<Buffer | string>
+  ): AsyncIterator<[string, string, boolean?]> {
+    let buf = "";
+    const eventBuf: SSE[] = [];
+
+    function parseEvent(eventStr: string) {
+      let data = "";
+      let id = "";
+      let type;
+      const lines = eventStr.split("\n").filter(Boolean);
+      const idRegEx = /^id: (.*?)$/;
+      const dataRegEx = /^data: (.*?)$/;
+      const typeRegEx = /^type: (.*?)$/;
+      for (const l of lines) {
+        const idRes = idRegEx.exec(l);
```

Comment:
Since these regexes are used only once, how do you feel about inlining them where they are used?